src/actions/web3.js

```diff
@@ -125,23 +125,19 @@ export const getLogs = address => async (dispatch, getState) => {
   });
 
   const allLogs = {};
-  const finalProcessor = () => {
+  const finalProcessor = async () => {
     if (Object.keys(allLogs).length > 0) {
       const statistics = getStatistics(icoConfig, allLogs);
       /* statistics array of two elements, index number 0 for statistcs, 
       index number 1 for csv content */
       dispatch(drawStatistics(statistics[0]));
       dispatch(allocateCSVFile(statistics[1]));
-
-      setCurrency('EUR', new Date(), (error, currencyResult) => {
-        if (error) {
-          dispatch({ type: 'SET_CURRENCY_ERROR', message: error });
-          return;
-        }
-        dispatch(setStatisticsByCurrency(currencyResult.currency,
-          currencyResult.value, currencyResult.time));
-        dispatch(showStatistics());
-      });
+      const baseCurrency = icoConfig.baseCurrency || 'ETH';
```

Comment:
move conversion rate computation outside of finalProcessor and make it non async again